packages/http/src/utils/__tests__/parseResponse.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { parseResponse, parseResponseBody, parseResponseHeaders } from '../parseResponse';
+import { assertLeft, assertRight } from '../../__tests__/utils';
+import { Headers } from 'node-fetch';
+
+describe('parseResponseBody()', () => {
+  describe('body is json', () => {
+    describe('body is parseable', () => {
+      it('returns parsed body', async () => {
+        const response = {
+          headers: new Map([['content-type', 'application/json']]),
+          json: jest.fn().mockResolvedValue({ test: 'test' }),
+          text: jest.fn(),
+        } as any;
+
+        assertRight(
+          await parseResponseBody(response)(),
+          body => expect(body).toEqual({ test: 'test' }),
+        );
+
+        expect(response.text).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('body is not parseable', () => {
+      it('returns error', async () => {
+        const response = {
+          headers: new Map([['content-type', 'application/json']]),
+          json: jest.fn().mockRejectedValue(new Error('Big Bada Boom')),
+          text: jest.fn(),
+        } as any;
+
+        assertLeft(
+          await parseResponseBody(response)(),
+          error => expect(error.message).toEqual('Big Bada Boom'),
+        );
+
+        expect(response.text).not.toHaveBeenCalled();
+      });
+    })
+  });
+
+  describe('body is not json', () => {
+    describe('body is readable', () => {
+      it('returns body text', async () => {
+        const response = {
+          headers: new Map([['content-type', 'text/html']]),
+          json: jest.fn(),
+          text: jest.fn().mockResolvedValue('<html>Test</html>'),
+        } as any;
```

Comment:
Instead of using the `as any` cast I would modify the `parseResponseBody` response to use `Pick`:

`response: Pick<Response, 'json' | 'text' | 'headers'>`

It might sound weird but I believe it's going to be way better than having a bunch of casts everywhere.